docs/ADR-18-content-mappings-flow-for-explorer-and-builder-in-world.md

```diff
@@ -0,0 +1,262 @@
+
+Examples can include: `asset.gltf` `assets/cube.glb` `assets/cube.png` etc.
+
+#### Asset Hash
+
+The hash used for composing the full url. This hash is usually assigned by the catalyst content server, and is guaranteed to be unique per individual asset.
+
+Any full asset url may look like e.g.: `https://peer.decentraland.org/contents/QmHash`
+
+### Content mappings and scene flow
+
+When a new scene is created, a message is sent from Kernel to Renderer specifying the data that's going to be needed to assemble a full asset url from a single local asset path. The data looks roughly like this:
+```
+{
+    // Here the value can vary depending on the connected realm
+    "baseUrl":"https://peer.decentraland.org/contents",
+
+    "contents":
+    {
+        "asset.gltf":"QmHash1",
+        "asset.png":"QmHash2",
+        "weapons/firearm1.gltf":"QmHash3"
+    } 
+}
+``` 
+
+This data is stored in a `ContentProvider` object [(see code).](https://github.com/decentraland/explorer/blob/b1545f8ae65d8e1198caeb18086fcde1d0be20f3/unity-client/Assets/Scripts/MainScripts/DCL/Controllers/Scene/ContentProvider/ContentProvider.cs#L9)
+
+Then, further down the line, a component can be sent with a payload that looks like this:
+```
+{
+    "src":"asset.gltf"
+}
+```
+
+In runtime we can receive this payload for any scene object. In order to download the object, we must resolve the full URL for it, like this:
+```
+baseUrl + resolved hash
+```
+The resolved hash is obtained querying a dictionary with the local asset path. 
+
+This dictionary is scoped to the scene. Meaning that any scene will have a specific dictionary of content mapping pairs and a determined `baseUrl`. The mapping pairs and the baseUrl are fetched from Kernel using our catalyst scene endpoints and then passed over to explorer's renderer.
+
+![resources/adr-18-content-mappings-flow-for-explorer-and-builder-in-world/fig-mappings-flow.svg](resources/adr-18-content-mappings-flow-for-explorer-and-builder-in-world/fig-mappings-flow.svg)
+<!--
+```sequence
+Catalyst->Kernel: Kernel picks realm url
+Catalyst->Kernel: Kernel fetches mapping pairs
+Kernel->Renderer: send LoadParcelScenes message\nwith baseUrl and content mappings
+Kernel->Renderer: create any sdk component\nwith src payload
+Renderer->Renderer: Use baseUrl and mappings\nto solve full asset url.
+```
+!-->
+
+This approach works fine for most of the cases. However, some issues can be observed:
+
+1) For big scenes, a pretty big content mappings payload have to be sent over to Renderer. This can produce performance hiccups.
+
+2) For content coming from dynamic sources (i.e.: builder or builder in world) the chance of hash collisions in the local asset path is high. Any artist can inadvertently name their assets using generic naming, and in this case a collision is going to occur.
```

Comment:
I would try rephrase (2) and (3) to make it clearer that:
* The main problem with the Builder In-World is that it adds assets dinamically, so our rigid mapping logic doesn't work anymore. This is the same, whether we support custom user assets or not
* The collision is in the local asset path, and not the hash itself. 
* When we support custom user assets, the chances for local asset path collisions becomes really high

Reply:
Edited to improve clarity on those points. Thanks!